Dockerfile

```diff
@@ -0,0 +1,72 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+ARG PG_MAJOR_VERSION=9.6
+ARG PG_MINOR_VERSION=9.6.18
+ARG BASE_IMAGE_VERSION=9.6.18
+
+ARG DATASKETCHES_CPP_VERSION=2.0.0-incubating
+
+FROM postgres:$BASE_IMAGE_VERSION
+
+MAINTAINER dev@datasketches.apache.org
+
+ARG PG_MAJOR_VERSION
+ARG PG_MINOR_VERSION
+ARG DATASKETCHES_CPP_VERSION
+
+ENV PG_MAJOR=$PG_MAJOR_VERSION
+ENV PG_MINOR=$PG_MINOR_VERSION
+ENV DATASKETCHES_CPP=$DATASKETCHES_CPP_VERSION
+
+
+ADD . /datasketches-postgresql
+WORKDIR /datasketches-postgresql
+
+RUN echo "===> Adding prerequisites..."                      && \
+    apt-get update -y                                        && \
+    DEBIAN_FRONTEND=noninteractive                              \
+        apt-get install --no-install-recommends --allow-downgrades -y -q \
+                ca-certificates                                 \
+                build-essential wget                            \
+                postgresql-server-dev-$PG_MAJOR=$PG_MINOR\*     \
+                libpq-dev=$PG_MINOR\* libpq5=$PG_MINOR\* libecpg-dev && \
+    \
+    \                
+    echo "===> Building datasketches..."                     && \
+    wget https://github.com/apache/incubator-datasketches-cpp/archive/$DATASKETCHES_CPP.tar.gz && \
```

Comment:
Downloading apache releases from github is not allowed.   And to do it automatically requires a script to balance between mirrors.  Please read [](https://infra.apache.org/release-download-pages).

Reply:
[https://infra.apache.org/release-download-pages](https://infra.apache.org/release-download-pages)

Reply:
I borrow the download code from [apache/fluo](https://github.com/apache/fluo-docker/blob/1ccfc40d652fc715dee33a563606014356cff595/Dockerfile#L40). It should fit the downloading policy.

Reply:
Not necessarily.   I would post this question on general@incubator.apache.org and see what the response is.  I have been told by very senior folks at ASF that there are many older Apache projects that violate a lot of ASF policies but they have been grandfathered in.  Also the policies themselves have changed over the years as ASF has grown.  So the best advice would be to ask some of the senior folks like Justin Mclean, Dave Fischer, etc.